boards/opentitan/earlgrey-cw310/Makefile

```diff
@@ -54,9 +54,9 @@ verilator: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/release/$(PLATFORM).bin
 		-within $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/release/$(PLATFORM).bin \
 		-binary -range-pad 8 --output binary.64.vmem --vmem 64
 	$(OPENTITAN_TREE)/build/lowrisc_dv_chip_verilator_sim_0.1/sim-verilator/Vchip_sim_tb \
-		--meminit=rom,$(OPENTITAN_TREE)/build-out/sw/device/lib/testing/test_rom/test_rom_sim_verilator.scr.39.vmem \
+		--meminit=rom,$(OPENTITAN_TREE)/bazel-out/k8-fastbuild-ST-97f470ee3b14/bin/sw/device/lib/testing/test_rom/test_rom_sim_verilator.scr.39.vmem \
```

Comment:
I would rather not call bazel directly here. Calling bazel is slow and just then required me to download and install something just to run this test.

By just using paths we can use a compile OT once and then run Tock often as you like. It also allows someone to use pre-built binaries if they want

Reply:
I should have been more precise: You _must not_ use `bazel-out` paths directly.  The path component `k8-fastbuild-ST-<hex>` is based on the toolchain selection and configuration.  It is not meant to be used directly and can change if we change something about the toolchain configuration.

The download is one-time setup: it downloads the version of bazel that is required by the opentitan project.  Bazel will not rebuild opentitan every time you call it - it is going to cache build outputs and will re-use them as needed.

Reply:
Is that actually an issue? As we're pinned to a particular version and build, this path will only change if/when we change OT images _and_ bazel happens to change its output path. Requiring the installation of a new tool just to compute a build path for this rare case feels like the wrong side of the tradeoff.

One core goal with this part of the infrastructure is to make testing Tock on the platform more accessible, i.e. it must work for folks who don't have an OT development environment set up.

Reply:
There two options here:
1. Operate with OPENTITAN_TREE and follow the guidance on using bazel.
2. Use a fixed release with well-known paths (and I mean a release artifact emitted by opentitan's build process -- not reaching into bazel-owned directory structure).

Personally, I'd prefer to be able to operate at opentitan@HEAD until opentitan stabilizes and we have a tag that represents a go-to-silicon release.  I understand that you may feel differently about that.

Reply:
The way Tock has historically operated w.r.t. opentitan was to consistently target a fixed OT commit, which by definition has well-known paths. If it were in fact possible to structure Tock's CI so as to enable us to consistently operate at opentitan@HEAD that might be nice to do, but things like changing register definitions and the potential for bugs in the opentitan repository could lead to changes in OT breaking Tock's QEMU CI that runs on OpenTitan, which would be pretty undesirable. Given the number of changes we regularly have to make to bump to a new opentitan version / release, having to change one set of build paths doesn't seem like a huge cost. 

@cfrantz perhaps you have a different view of how much will need to change (in tock/tock) across OT updates moving forward? I understand the autogenerated peripheral registers helps a good bit, but it seems like other changes (such as the manifest and pmp changes in this PR) are likely to continue requiring substantial adjustments anytime someone wants to try running Tock on a newer OT version.

Reply:
We tried running with OpenTitan HEAD early on. There are just too many breaking changes that don't get communicated back to us. As you can already see, it has taken weeks to bump the SHA (this PR still doesn't work correctly).

We also can't use the release artifacts, as they are deleted from the CI too quickly.

We should stick to the current strategy of known good releases. Which means we can use a fixed build path to avoid installing tools just to query a build path

Reply:
Heh, of course running from HEAD in the early days of the project was going to be difficult :)  Things were chaotic in the early days because nothing was defined yet.  These days, we're working towards tape-out of first silicon.

The artifacts published be the CI _are not_ release artifacts.  They are debug artifacts and don't have any guarantees about their lifetime.  And to quibble about terms, selecting opentitan@<hash> is also not a known-good release.  Unfortunately, I can't really make any good statement or recommendation about the current opentitan releases as the last thing released (as a formal release via github) is from 2019.

I am currently in the process or remedying this situation and I hope to have the project performing a formal release via an automated process on a regular cadence.  The formal release will contain bitstreams, rom images, the verilator model and opentitantool.

Reply:
Maybe it makes sense to keep the hardcoded paths for now, then switch over to formal release images when those images are available?